resources/helm/dask-gateway/templates/gateway/deployment.yaml

```diff
@@ -57,3 +57,25 @@ spec:
           ports:
             - containerPort: 8000
               name: api
+
+          {{- if .Values.gateway.livenessProbe.enabled }}
+          livenessProbe:
+            httpGet:
+              path: /api/health
+              port: api
+            initialDelaySeconds: {{ .Values.gateway.livenessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.gateway.livenessProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.gateway.livenessProbe.timeoutSeconds }}
+            failureThreshold: {{ .Values.gateway.livenessProbe.failureThreshold }}
```

Comment:
To reduce the issues of working with Helm's templates, I suggest opting for doing something more like.

```yaml
{{- with .Values.gateway.livenessProbe }}
livenessProbe:
  {{- .Values.gateway.livenessProbe | toYaml | nindent 12 }}
{{- end }}
```